clippy_utils/src/lib.rs

```diff
@@ -1224,6 +1224,57 @@ pub fn get_parent_as_impl(tcx: TyCtxt<'_>, id: HirId) -> Option<&Impl<'_>> {
     }
 }
 
+/// Remove blocks around an expression.
+///
+/// Ie. `x`, `{ x }` and `{{{{ x }}}}` all give `x`. `{ x; y }` and `{}` return
+/// themselves.
```

Comment:
The implementation will also halt at blocks which contain a single statement. I think it's worth to mention this and provide an example. Also, to make clear what distinguishes this function from the other one. It would also be good to mention that this only peels normal blocks.

Maybe a table or something might be better than a sentence for the examples, even if this is still not perfect.

```rs
/// | Input           | Output        |
/// |-----------------|---------------|
/// | `x`             | `x`           |
/// | `{ x }`         | `x`           |
/// | `{{{{ x }}}}`   | `x`           |
/// | `{}`            | `{}`          |
/// | `{ y; x }`      | `{ y; x }`    |
/// | `{ y; }`        | `{ y; }`      |
/// | `{ {x}; }`      | `{ {x}; }`    |
/// | `{ unsafe {x} }`| `unsafe {x} ` |
```